dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/AbstractRegistry.java

```diff
@@ -119,15 +121,15 @@ protected void setUrl(URL url) {
     }
 
     public Set<URL> getRegistered() {
-        return registered;
+        return Collections.unmodifiableSet(registered);

```

Comment:
@kezhenxu94  thanks for this PR. As you are returning the copy of collection, i think you can avoid making it unmodifiable.

Reply:
Hi @khanimteyaz , it's just a wrapper, not a copy, so it's OK here :)